### PR TITLE
test: add new ann scenarios to the python benchmarks

### PR DESCRIPTION
**Pull request by westonpace**
This adds test datasets with deletion vectors and with new unindexed fragments

### CODE REVIEW

**Comment by westonpace** [comment]
Example Output:

```
---------------------------------------------------------------------------------------- benchmark 'query_ann': 9 tests ----------------------------------------------------------------------------------------
Name (time in ms)                                   Min                 Max               Mean            StdDev             Median               IQR            Outliers      OPS            Rounds  Iterations
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_flat_index_search[with_new_rows]           56.7829 (1.0)       61.7166 (1.0)      59.4567 (1.0)      1.4351 (3.50)     59.8184 (1.0)      1.0468 (1.57)          6;4  16.8189 (1.0)          15           1
test_flat_index_search[clean]                   60.2991 (1.06)      62.0837 (1.01)     60.9050 (1.02)     0.5336 (1.30)     60.9220 (1.02)     0.7441 (1.12)          4;0  16.4190 (0.98)         13           1
test_ivf_pq_index_search[clean]                 71.3794 (1.26)      72.6038 (1.18)     71.9130 (1.21)     0.4105 (1.0)      71.7871 (1.20)     0.6651 (1.0)           5;0  13.9057 (0.83)         14           1
test_ivf_pq_index_search[with_new_rows]         72.3877 (1.27)      77.5621 (1.26)     74.7442 (1.26)     2.0292 (4.94)     74.2366 (1.24)     4.0826 (6.14)          7;0  13.3790 (0.80)         14           1
test_flat_index_search[with_delete_files]       76.2899 (1.34)      77.9860 (1.26)     77.0855 (1.30)     0.5323 (1.30)     77.2210 (1.29)     0.8808 (1.32)          4;0  12.9726 (0.77)         11           1
test_knn_search[clean]                          83.7363 (1.47)      96.2798 (1.56)     90.0815 (1.52)     4.1720 (10.16)    90.5738 (1.51)     7.0653 (10.62)         5;0  11.1011 (0.66)         11           1
test_knn_search[with_delete_files]              86.6546 (1.53)      99.3434 (1.61)     92.7905 (1.56)     3.4818 (8.48)     91.2431 (1.53)     4.2368 (6.37)          3;0  10.7770 (0.64)         12           1
test_ivf_pq_index_search[with_delete_files]     93.2096 (1.64)      94.9831 (1.54)     93.8758 (1.58)     0.5133 (1.25)     93.7430 (1.57)     0.6878 (1.03)          3;0  10.6524 (0.63)         11           1
test_knn_search[with_new_rows]                  94.0760 (1.66)     100.0762 (1.62)     96.3771 (1.62)     2.0682 (5.04)     95.4485 (1.60)     2.9146 (4.38)          3;0  10.3759 (0.62)         10           1
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

File: python/python/benchmarks/test_search.py [hunk 1]
```diff
@@ -57,6 +69,60 @@ def test_dataset(data_dir: Path) -> lance.LanceDataset:
...
+@pytest.fixture(scope="module")
+def datasets(data_dir: Path) -> Datasets:
+    return Datasets(
+        clean=create_base_dataset(data_dir),
+        with_delete_files=create_delete_dataset(data_dir),
+        with_new_rows=create_new_rows_dataset(data_dir),
+    )
+
+
+@pytest.fixture(scope="module", params=Datasets._fields)
```
Comment (wjones127) [inline python/python/benchmarks/test_search.py:121]: Had no idea you could do that. Nice!

File: python/python/benchmarks/test_search.py [hunk 2]
```diff
@@ -57,6 +69,60 @@ def test_dataset(data_dir: Path) -> lance.LanceDataset:
...
+    shutil.copytree(clean_path, tmp_path)
+
+    dataset = lance.dataset(tmp_path)
+    dataset.delete("filterable % 2 != 0")
+
+    return dataset
+
+
+def create_new_rows_dataset(data_dir):
+    tmp_path = data_dir / "search_dataset_with_new_rows"
```
Comment (wjones127) [inline python/python/benchmarks/test_search.py:88]: In the future, we might consider just making each of these a different version of the same dataset. That way we don't have to manage all these directories.

File: python/python/benchmarks/test_search.py [hunk 3]
```diff
@@ -57,6 +69,60 @@ def test_dataset(data_dir: Path) -> lance.LanceDataset:
...
+    tmp_path = data_dir / "search_dataset_with_delete"
+    dataset = find_or_clean(tmp_path)
+    if dataset:
+        return dataset
+
+    clean_path = data_dir / "search_dataset"
+    shutil.copytree(clean_path, tmp_path)
+
+    dataset = lance.dataset(tmp_path)
+    dataset.delete("filterable % 2 != 0")
```
Comment (wjones127) [inline python/python/benchmarks/test_search.py:82]: Something to consider for the future: we should split deletion into 2 x 2 matrix with along `contiguous vs random` and `deletion %: 2% vs 20%`.

**Review by wjones127 — APPROVED** [review]

**Pull request merged**